examples/src/test/java/jdocs/guide/EventGeneratorApp.java

```diff
@@ -0,0 +1,238 @@
+/*
+ * Copyright (C) 2020 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+// #guideEventGeneratorApp
+package jdocs.guide;
+
+import akka.actor.typed.ActorRef;
+import akka.actor.typed.ActorSystem;
+import akka.actor.typed.Behavior;
+import akka.actor.typed.javadsl.Behaviors;
+import akka.cluster.sharding.typed.javadsl.ClusterSharding;
+import akka.cluster.sharding.typed.javadsl.Entity;
+import akka.cluster.sharding.typed.javadsl.EntityTypeKey;
+import akka.cluster.typed.Cluster;
+import akka.cluster.typed.Join;
+import akka.cluster.typed.SelfUp;
+import akka.cluster.typed.Subscribe;
+import akka.persistence.typed.PersistenceId;
+import akka.persistence.typed.javadsl.CommandHandler;
+import akka.persistence.typed.javadsl.EventHandler;
+import akka.persistence.typed.javadsl.EventSourcedBehavior;
+import akka.stream.javadsl.Sink;
+import akka.stream.javadsl.Source;
+import com.typesafe.config.Config;
+import com.typesafe.config.ConfigFactory;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.time.Duration;
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+import java.util.UUID;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+import java.util.stream.Stream;
+
+/**
+ * Generate a shopping cart every 1 second and check it out. Each cart will contain a variety of
+ * `ItemAdded`, `ItemQuantityAdjusted` and `ItemRemoved` events preceding the the cart `Checkout`
+ * event.
+ */
+public class EventGeneratorApp {
+  public static void main(String[] args) throws Exception {
+    Boolean clusterMode = (args.length > 0 && args[0].equals("cluster"));
+    Config config = config();
+    ActorSystem<Guardian.Command> system =
+        ActorSystem.create(Guardian.create(clusterMode), "EventGeneratorApp", config);
+  }
+
+  private static Config config() {
+    return ConfigFactory.parseString("akka.actor.provider = \"cluster\"\n")
+        .withFallback(ConfigFactory.load("guide-shopping-cart-app.conf"));
+  }
+}
+
+class Guardian {
+  interface Command {}
+
+  static final class Start implements Command {}
+
+  static final List<String> Products =
+      List.of("cat t-shirt", "akka t-shirt", "skis", "bowling shoes");
+
+  static final int MaxQuantity = 5;
+  static final int MaxItems = 3;
+  static final int MaxItemsAdjusted = 3;
+
+  static final EntityTypeKey<ShoppingCartEvents.Event> EntityKey =
+      EntityTypeKey.create(ShoppingCartEvents.Event.class, "shopping-cart-event");
+
+  static Behavior<Command> create(Boolean clusterMode) {
+
+    return Behaviors.setup(
+        context -> {
+          ActorSystem<Void> system = context.getSystem();
+          Cluster cluster = Cluster.get(system);
+
+          ActorRef<SelfUp> upAdapter = context.messageAdapter(SelfUp.class, up -> new Start());
```

Comment:
It's not necessary to wait until Up. Can be simplified by removing this and the Start message.

Reply:
:+1: